File: GitNexus/gitnexus/test/fixtures/lang-resolution/php-deep-field-chain/Service.php
<?php

class Service {
    public function processUser(User $user): void {
        // 2-level chain: $user->address → Address, then ->save() → Address#save
        $user->address->save();

        // 3-level chain: $user->address → Address, ->city → City, ->getName() → City#getName
        $user->address->city->getName();
    }
}

File: GitNexus/gitnexus/test/fixtures/lang-resolution/csharp-deep-field-chain/Service.cs
namespace DeepFieldChain;

public class Service
{
    public static void ProcessUser(User user)
    {
        // 2-level chain: user.Address → Address, then .Save() → Address#Save
        user.Address.Save();

        // 3-level chain: user.Address → Address, .City → City, .GetName() → City#GetName
        user.Address.City.GetName();
    }
}
